[CI] WriteLoadConstraintDeciderIT testAverageWriteLoadMetricIsPublished failing

**Build Scans:**
- [elasticsearch-intake #37190 / part1](https://gradle-enterprise.elastic.co/s/pctr4nwkoebpw)
- [elasticsearch-pull-request #135117 / part-1](https://gradle-enterprise.elastic.co/s/4a7o3fkhsv4nm)

**Reproduction Line:**
```
./gradlew ":server:internalClusterTest" --tests "org.elasticsearch.cluster.routing.allocation.decider.WriteLoadConstraintDeciderIT.testAverageWriteLoadMetricIsPublished" -Dtests.seed=C799653A1BA6BF81 -Dtests.locale=ksh-Latn-DE -Dtests.timezone=Cuba -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.cluster.routing.allocation.decider.WriteLoadConstraintDeciderIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testAverageWriteLoadMetricIsPublished'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: a collection with size <3>
     but: collection size was <2>
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testAverageWriteLoadMetricIsPublished
- [main] 1 consecutive failures in step part1
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 2 failures in test testAverageWriteLoadMetricIsPublished (22.2% fail rate in 9 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -246,9 +246,6 @@ tests:
 - class: org.elasticsearch.repositories.gcs.GoogleCloudStorageServiceTests
   method: testApplicationDefaultCredentialsFallbackToComputeEngine
   issue: https://github.com/elastic/elasticsearch/issues/145689
-- class: org.elasticsearch.cluster.routing.allocation.decider.WriteLoadConstraintDeciderIT
-  method: testAverageWriteLoadMetricIsPublished
-  issue: https://github.com/elastic/elasticsearch/issues/145855
 - class: org.elasticsearch.xpack.ccr.CcrSeqNoPruningIT
   method: testSeqNoPrunedOnLeaderAfterFollowerCatchesUp
   issue: https://github.com/elastic/elasticsearch/issues/145877

--- a/server/src/internalClusterTest/java/org/elasticsearch/cluster/routing/allocation/decider/WriteLoadConstraintDeciderIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/cluster/routing/allocation/decider/WriteLoadConstraintDeciderIT.java
@@ -659,6 +659,7 @@ public class WriteLoadConstraintDeciderIT extends ESIntegTestCase {
             .put(onlyRoles(Set.of(DiscoveryNodeRole.MASTER_ROLE, DiscoveryNodeRole.INDEX_ROLE)))
             .build();
         final var dataNodes = internalCluster().startNodes(3, settings);
+        ensureStableCluster(3);
 
         // Refresh cluster info (should trigger polling)
         refreshClusterInfo();

